activesupport/lib/active_support/cache.rb

```diff
@@ -547,6 +579,10 @@ def normalize_key(key, options)
           key
         end
 
+        def normalize_version(key, options)
+          options[:version] || key.try(:cache_version)
```

Comment:
You need to support more advanced array  keys like:

``` ruby
fetch(["preview", post])
fetch(["preview", post, post.author])
```

There is an implementation in `expand_cache_version` here: https://github.com/rails/rails/pull/29107/files#diff-438394335b9c1ce6ec4f67a407f50a42R89